File: weave-cookbook/python/examples/shared/catalog.py
"""Load and filter the Game Night board game catalog."""

import json
from pathlib import Path

CATALOG_PATH = Path(__file__).parent / "data" / "games.json"


def load_catalog() -> list[dict]:
    return json.loads(CATALOG_PATH.read_text())


def filter_games(
    games: list[dict],
    *,
    players: int,
    max_minutes: int,
    difficulty: str | None = None,
) -> list[dict]:
    """Return games that fit the player count, time budget, and optional difficulty."""
    return [
        game
        for game in games
        if game["min_players"] <= players <= game["max_players"]
        and game["playtime_minutes"] <= max_minutes
        and (difficulty is None or game["difficulty"] == difficulty)
    ]


def describe_game(game: dict) -> str:
    return (
        f"- {game['name']} ({game['min_players']}-{game['max_players']} players, "
        f"{game['playtime_minutes']} min, {game['difficulty']}): {game['description']}"
    )

File: weave-cookbook/typescript/examples/shared/catalog.ts
// Load and filter the Game Night board game catalog.

import { readFileSync } from 'node:fs';

export type Difficulty = 'light' | 'medium' | 'heavy';

export interface Game {
  name: string;
  min_players: number;
  max_players: number;
  playtime_minutes: number;
  difficulty: Difficulty;
  tags: string[];
  description: string;
}

export interface FilterOptions {
  players: number;
  maxMinutes: number;
  difficulty?: Difficulty;
}

const CATALOG_URL = new URL('./data/games.json', import.meta.url);

export function loadCatalog(): Game[] {
  return JSON.parse(readFileSync(CATALOG_URL, 'utf8'));
}

export function filterGames(games: Game[], options: FilterOptions): Game[] {
  return games.filter(
    (game) =>
      game.min_players <= options.players &&
      options.players <= game.max_players &&
      game.playtime_minutes <= options.maxMinutes &&
      (options.difficulty === undefined || game.difficulty === options.difficulty),
  );
}

export function describeGame(game: Game): string {
  return (
    `- ${game.name} (${game.min_players}-${game.max_players} players, ` +
    `${game.playtime_minutes} min, ${game.difficulty}): ${game.description}`
  );
}
